src/components/notification/process-server-status.ts

```diff
@@ -28,6 +28,8 @@ import { Probe } from '../../interfaces/probe'
 import { ServerAlertState } from '../../interfaces/probe-status'
 import { ValidatedResponse } from '../../plugins/validate-response'
 
+let isFirstTimeSendEvent = true
```

Comment:
Just realized something. I think there's a bug in this implementation. I think it only sends the event for the first probe. Instead of adding isFirstTimeSendEvent variable here, maybe you should create another state machine to keep the isFirstTimeSendEvent state for all the probes.

Reply:
Solved!